routing/bicycle_directions.cpp

```diff
@@ -170,7 +185,18 @@ void BicycleDirectionsEngine::Generate(IRoadGraph const & graph, vector<Junction
   Route::TTimes turnAnnotationTimes;
   Route::TStreets streetNames;
 
-  MakeTurnAnnotation(resultGraph, delegate, routeGeometry, turns, turnAnnotationTimes, streetNames);
+  MakeTurnAnnotation(resultGraph, delegate, routeGeometry, turns, turnAnnotationTimes,
+                     streetNames, trafficSegs);
+  // @TODO(bykoianko) The invariant below it's an issue but now it's so and it should be checked.
+  // The problem is every edge is added as a pair of points to route geometry.
+  // So all the points except for beginning and ending are duplicated. It should
+  // be fixed in the future.
+
+  // Note 1. Accoding to current implementation all internal points are duplicated for
+  // all A* routes.
+  // Note 2. Number of |trafficSegs| should be equal to number of |routeEdges|.
+  ASSERT_EQUAL(routeGeometry.size(), 2 * (pathSize - 2) + 2, ());
```

Comment:
I guess ```2 * (n - 2) + 2``` is more readable, as it means that ```each non-end point is duplicated, end points are left as is```, what is the same as what is specified in the comment above.